Move UIKit SPI used only for testing into separate header

whsieh opened the pull request:
#### ac6be1003b5936badc0a7333bfa630df2a8f498c
<pre>
Move UIKit SPI used only for testing into separate header
<a href="https://bugs.webkit.org/show_bug.cgi?id=260510">https://bugs.webkit.org/show_bug.cgi?id=260510</a>
rdar://114242032

Reviewed by Megan Gardner.

Move TestWebKitAPI&apos;s existing `Tools/TestWebKitAPI/ios/UIKitSPI.h` into `TestRunnerShared/`, as
`Tools/TestRunnerShared/spi/UIKitSPIForTesting.h`; use this on iOS in WebKitTestRunner,
DumpRenderTree and TestWebKitAPI by augmenting header search paths to include
`Tools/TestRunnerShared/spi/*`.

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:
* Tools/DumpRenderTree/DumpRenderTree.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/ios/DumpRenderTreeBrowserView.h:
* Tools/DumpRenderTree/ios/PixelDumpSupportIOS.mm:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
* Tools/DumpRenderTree/mac/DumpRenderTreeWindow.h:
* Tools/DumpRenderTree/mac/TestRunnerMac.mm:
* Tools/TestRunnerShared/cocoa/LayoutTestSpellChecker.mm:
* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h: Renamed from Tools/TestWebKitAPI/ios/UIKitSPI.h.
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit/AdvancedPrivacyProtections.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ClipboardTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/CopyURL.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DocumentEditingContext.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EditorStateTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FontAttributes.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ImageAnalysisTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/InsertTextAlternatives.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PasteboardUtilities.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SystemColors.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAppHighlights.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKContentViewEditingActions.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewEditActions.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewFindString.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewGetContents.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSStylusSupport.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/DateTimeInputsAccessoryViewTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/ScrollToRevealSelection.mm:
* Tools/TestWebKitAPI/Tests/ios/ActionSheetTests.mm:
* Tools/TestWebKitAPI/Tests/ios/AutocorrectionTestsIOS.mm:
* Tools/TestWebKitAPI/Tests/ios/CustomContentViewGestures.mm:
* Tools/TestWebKitAPI/Tests/ios/DragAndDropTestsIOS.mm:
* Tools/TestWebKitAPI/Tests/ios/EnterKeyHintTests.mm:
* Tools/TestWebKitAPI/Tests/ios/FocusPreservationTests.mm:
* Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
* Tools/TestWebKitAPI/Tests/ios/NavigationSwipeTests.mm:
* Tools/TestWebKitAPI/Tests/ios/OverflowScrollViewTests.mm:
* Tools/TestWebKitAPI/Tests/ios/ScrollViewInsetTests.mm:
* Tools/TestWebKitAPI/Tests/ios/SelectionByWord.mm:
* Tools/TestWebKitAPI/Tests/ios/SelectionModifyByParagraphBoundary.mm:
* Tools/TestWebKitAPI/Tests/ios/TextAlternatives.mm:
* Tools/TestWebKitAPI/Tests/ios/TextAutosizingBoost.mm:
* Tools/TestWebKitAPI/Tests/ios/TextServicesTests.mm:
* Tools/TestWebKitAPI/Tests/ios/TextStyleFontSize.mm:
* Tools/TestWebKitAPI/Tests/ios/TouchEventTests.mm:
* Tools/TestWebKitAPI/Tests/ios/UIPasteboardTests.mm:
* Tools/TestWebKitAPI/Tests/ios/UIWKInteractionViewProtocol.mm:
* Tools/TestWebKitAPI/Tests/ios/UserInterfaceIdiomUpdate.mm:
* Tools/TestWebKitAPI/Tests/ios/WKScrollViewTests.mm:
* Tools/TestWebKitAPI/Tests/ios/WKWebViewAutofillTests.mm:
* Tools/TestWebKitAPI/cocoa/DragAndDropSimulator.h:
* Tools/TestWebKitAPI/cocoa/TestContextMenuDriver.mm:
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
* Tools/TestWebKitAPI/ios/DragAndDropSimulatorIOS.mm:
* Tools/TestWebKitAPI/ios/UserInterfaceSwizzler.h:
* Tools/WebKitTestRunner/Configurations/Base.xcconfig:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.mm:
* Tools/WebKitTestRunner/ios/GeneratedTouchesDebugWindow.mm:
* Tools/WebKitTestRunner/ios/HIDEventGenerator.h:
* Tools/WebKitTestRunner/ios/HIDEventGenerator.mm:
* Tools/WebKitTestRunner/ios/PlatformWebViewIOS.mm:
* Tools/WebKitTestRunner/ios/TestControllerIOS.mm:
* Tools/WebKitTestRunner/ios/UIKitSPIForTesting.h: Removed.
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
* Tools/WebKitTestRunner/ios/mainIOS.mm:

Canonical link: <a href="https://commits.webkit.org/268564@main">https://commits.webkit.org/268564@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b90fc05821eef647f1eaa2df339a0771c258c2ff

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20048 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20487 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21105 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21947 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18715 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20278 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23729 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20645 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20268 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20195 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17424 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22797 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17365 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18221 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/24475 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18442 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18397 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22465 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18990 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/16112 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18185 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4804 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22530 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18816 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
This is great! Is it possible to restrict visibility of the WebKit/Platform/spi/ios/UIKitSPI.h header from Tools/ now?

Also, does WebCore/PAL/pal/spi/ios/UIKitSPI.h need a similar treatment?

whsieh committed:
Move UIKit SPI used only for testing into separate header
https://bugs.webkit.org/show_bug.cgi?id=260510
rdar://114242032

Reviewed by Megan Gardner.

Move TestWebKitAPI's existing `Tools/TestWebKitAPI/ios/UIKitSPI.h` into `TestRunnerShared/`, as
`Tools/TestRunnerShared/spi/UIKitSPIForTesting.h`; use this on iOS in WebKitTestRunner,
DumpRenderTree and TestWebKitAPI by augmenting header search paths to include
`Tools/TestRunnerShared/spi/*`.

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:
* Tools/DumpRenderTree/DumpRenderTree.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/ios/DumpRenderTreeBrowserView.h:
* Tools/DumpRenderTree/ios/PixelDumpSupportIOS.mm:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
* Tools/DumpRenderTree/mac/DumpRenderTreeWindow.h:
* Tools/DumpRenderTree/mac/TestRunnerMac.mm:
* Tools/TestRunnerShared/cocoa/LayoutTestSpellChecker.mm:
* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h: Renamed from Tools/TestWebKitAPI/ios/UIKitSPI.h.
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit/AdvancedPrivacyProtections.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ClipboardTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/CopyURL.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DocumentEditingContext.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EditorStateTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FontAttributes.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ImageAnalysisTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/InsertTextAlternatives.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PasteboardUtilities.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SystemColors.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAppHighlights.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKContentViewEditingActions.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewEditActions.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewFindString.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewGetContents.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSStylusSupport.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/DateTimeInputsAccessoryViewTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/ScrollToRevealSelection.mm:
* Tools/TestWebKitAPI/Tests/ios/ActionSheetTests.mm:
* Tools/TestWebKitAPI/Tests/ios/AutocorrectionTestsIOS.mm:
* Tools/TestWebKitAPI/Tests/ios/CustomContentViewGestures.mm:
* Tools/TestWebKitAPI/Tests/ios/DragAndDropTestsIOS.mm:
* Tools/TestWebKitAPI/Tests/ios/EnterKeyHintTests.mm:
* Tools/TestWebKitAPI/Tests/ios/FocusPreservationTests.mm:
* Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
* Tools/TestWebKitAPI/Tests/ios/NavigationSwipeTests.mm:
* Tools/TestWebKitAPI/Tests/ios/OverflowScrollViewTests.mm:
* Tools/TestWebKitAPI/Tests/ios/ScrollViewInsetTests.mm:
* Tools/TestWebKitAPI/Tests/ios/SelectionByWord.mm:
* Tools/TestWebKitAPI/Tests/ios/SelectionModifyByParagraphBoundary.mm:
* Tools/TestWebKitAPI/Tests/ios/TextAlternatives.mm:
* Tools/TestWebKitAPI/Tests/ios/TextAutosizingBoost.mm:
* Tools/TestWebKitAPI/Tests/ios/TextServicesTests.mm:
* Tools/TestWebKitAPI/Tests/ios/TextStyleFontSize.mm:
* Tools/TestWebKitAPI/Tests/ios/TouchEventTests.mm:
* Tools/TestWebKitAPI/Tests/ios/UIPasteboardTests.mm:
* Tools/TestWebKitAPI/Tests/ios/UIWKInteractionViewProtocol.mm:
* Tools/TestWebKitAPI/Tests/ios/UserInterfaceIdiomUpdate.mm:
* Tools/TestWebKitAPI/Tests/ios/WKScrollViewTests.mm:
* Tools/TestWebKitAPI/Tests/ios/WKWebViewAutofillTests.mm:
* Tools/TestWebKitAPI/cocoa/DragAndDropSimulator.h:
* Tools/TestWebKitAPI/cocoa/TestContextMenuDriver.mm:
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
* Tools/TestWebKitAPI/ios/DragAndDropSimulatorIOS.mm:
* Tools/TestWebKitAPI/ios/UserInterfaceSwizzler.h:
* Tools/WebKitTestRunner/Configurations/Base.xcconfig:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.mm:
* Tools/WebKitTestRunner/ios/GeneratedTouchesDebugWindow.mm:
* Tools/WebKitTestRunner/ios/HIDEventGenerator.h:
* Tools/WebKitTestRunner/ios/HIDEventGenerator.mm:
* Tools/WebKitTestRunner/ios/PlatformWebViewIOS.mm:
* Tools/WebKitTestRunner/ios/TestControllerIOS.mm:
* Tools/WebKitTestRunner/ios/UIKitSPIForTesting.h: Removed.
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
* Tools/WebKitTestRunner/ios/mainIOS.mm:

Canonical link: https://commits.webkit.org/268564@main
